qiskit/algorithms/fidelities/fidelity.py

```diff
@@ -0,0 +1,146 @@
+        """
+        circuit = circuit_1.compose(circuit_2.inverse())
+        circuit.measure_all()
+        return circuit
+
+    def _run(
+        self,
+        circuits_1: Sequence[QuantumCircuit],
+        circuits_2: Sequence[QuantumCircuit],
+        values_1: Sequence[Sequence[float]] | None = None,
+        values_2: Sequence[Sequence[float]] | None = None,
+        **run_options,
+    ) -> FidelityResult:
+        r"""
+        Compute the state overlap (fidelity) calculation between 2
+        parametrized circuits (left and right) for a specific set of parameter
+        values (left and right) following the compute-uncompute method, where
+        the fidelity corresponds to:
+
+            :math:`|\langle\psi(x)|\phi(y)\rangle|^2`
+
+        Args:
+            circuits_1: (Parametrized) quantum circuits preparing :math:`|\psi\rangle`.
+            circuits_2: (Parametrized) quantum circuits preparing :math:`|\phi\rangle`.
+            values_1: Numerical parameters to be bound to the first circuits.
+            values_2: Numerical parameters to be bound to the second circuits.
+            run_options: Backend runtime options used for circuit execution.
+
+        Returns:
+            The result of the fidelity calculation.
+
+        Raises:
+            ValueError: At least one pair of circuits must be defined.
+        """
+
+        self._preprocess_inputs(circuits_1, circuits_2, values_1, values_2)
+
+        if len(self._circuits) == 0:
+            raise ValueError(
+                "At least one pair of circuits must be defined to calculate the state overlap."
+            )
+
+        # The priority of run options is as follows:
+        # run_options in `evaluate` method > fidelity's default run_options >
+        # primitive's default run_options.
+        run_options = run_options or self._default_run_options
+
+        if len(self._parameter_values) > 0:
+            job = self._sampler.run(
+                circuits=self._circuits, parameter_values=self._parameter_values, **run_options
+            )
+        else:
+            job = self._sampler.run(circuits=self._circuits, **run_options)
+
+        result = job.result()
+
+        # if error mitigation is added in the future, we will have to handle
+        # negative values in some way (e.g. clipping to zero)
+        overlaps = [prob_dist.get(0, 0) for prob_dist in result.quasi_dists]
+
+        return FidelityResult(values=overlaps, metadata=run_options)
+
+    def run(
+        self,
+        circuits_1: Sequence[QuantumCircuit],
+        circuits_2: Sequence[QuantumCircuit],
+        values_1: Sequence[Sequence[float]] | None = None,
+        values_2: Sequence[Sequence[float]] | None = None,
+        **run_options,
+    ) -> PrimitiveJob:
+        r"""
+        Run asynchronously the state overlap (fidelity) calculation between 2
+        parametrized circuits (left and right) for a specific set of parameter
```

Comment:
Should parametrized go in brackets here like  other places to be consistent - I take that to mean it may or may not be parametrized.